### DISCUSSION TITLE
What to do when a channel is closed

### DISCUSSION THREAD

**Discussion by rvkennedy** [paullouisageneau/libdatachannel#835] [Q&A]
I often see a situation where my data channels close. It's unclear why this happens, but I want to reopen them to continue the connection.
What is the canonical way to do this? Should I create a new channel? If so, should it be distinct in id and/or label from the closed one?
Or is there a way to reopen a closed channel?

**Comment by paullouisageneau** [comment]
A data channel can be closed for two reasons: either the remote peer closed the data channel, or the peer connection it self was closed or lost connectivity. You are probably in the second case, in that case you should just re-open a new peer connection.

**Reply by rvkennedy** [reply]
Thanks Paul Louis. So if any channel becomes closed without my explicit instruction, I should expect that all channels of a connection will close?
And I should create a new peer connection rather than attempting to add channels to the existing connection with createDataChannel?

**Reply by paullouisageneau** [reply]
If you never close the data channel yourself on one side or the other, you may assume the connection is lost if it closes. In the general case, you should actually react to the peer connection state, you can observe it with the `PeerConnection::onStateChange` callback. If the peer connection is disconnected or failed, you should not create new data channels, instead you should indeed re-create a new peer connection.

**Reply by rvkennedy** [reply]
Will do. Thanks!
